pkg/cri/resource-manager/policy/builtin/topology-aware/resources.go

```diff
@@ -830,6 +839,73 @@ func (cs *supply) DumpAllocatable() string {
 	return allocatable
 }
 
+// prettyMem formats the given amount as k, M, G, or T units.
+func prettyMem(amount uint64) string {
+	units := []string{"k", "M", "G", "T"}
+	a := float64(amount)
+	f := a
+	u := ""
+	for _, unit := range units {
+		if f /= float64(1024); f < 1 {
+			break
+		}
```

Comment:
Do I read this correctly that now we only get pretty values less than 1. I think we should check for <1024 so that we'd get values between [0, 1024[ (?)

Also a super nit^10 that does not really matter: instead of consecutive floating point divisions I'd just compare integer values and only do the division when we find the limit

Reply:
> Do I read this correctly that now we only get pretty values less than 1. I think we should check for <1024 so that we'd get values between [0, 1024[ (?)

No, that's exactly the range. We stop dividing once the result goes below 1, but the value used is from the previous iteration of the loop. You get logs like this:
```
D: [     policy     ] <post-alloc>      - DRAM: free: 108.58G, granted 80.00G
D: [     policy     ] <post-alloc>      - total free: 108.58G, total granted 80.00G
D: [     policy     ] <post-alloc>      - extra reservations:
D: [     policy     ] <post-alloc>        - coredns-f9fd979d6-w4qpw:coredns: 170.00M (DRAM 170.00M)
D: [     policy     ] <post-alloc>        - coredns-f9fd979d6-4g982:coredns: 170.00M (DRAM 170.00M)
```

Reply:
True, I prolly misread the `'f'` below in `FormatFloat` as `f` or smth

Reply:
Updated to do the conversion with comparisons and a single float64 division.